repos/spack_repo/builtin/packages/hdf5/package.py

```diff
@@ -552,6 +547,22 @@ def cmake_args(self):
             self.define_from_variant("HDF5_BUILD_TOOLS", "tools"),
         ]
 
+        # Always enable this option. This does not actually enable any
+        # features: it only *allows* the user to specify certain
+        # combinations of other arguments.
+        # The option name has been changed to HDF5_ALLOW_UNSUPPORTED for
+        # HDF5 2.0 and higher versions.
+        if self.spec.satisfies("@2.0.0:"):
+            args.append(self.define("HDF5_ALLOW_UNSUPPORTED", True))
+        else:
+            args.append(self.define("ALLOW_UNSUPPORTED", True))
+
+        # The name of option Z_LIB_SUPPORT was also changed to ZLIB_SUPPORT.
+        if self.spec.satisfies("@2.0.0:"):
+            args.append(self.define("HDF5_ENABLE_ZLIB_SUPPORT", True))
+        else:
+            args.append(self.define("HDF5_ENABLE_Z_LIB_SUPPORT", True))
+
         api = spec.variants["api"].value
         if api != "default":
             args.append(self.define("DEFAULT_API_VERSION", api))
```

Comment:
DEFAULT_API_VERSION was changed to HDF5_DEFAULT_API_VERSION. So this needs to be updated (not tested):

```
api = spec.variants["api"].value
if api != "default":
    if self.spec.satisfies("@2.0.0:"):
        args.append(self.define("HDF5_DEFAULT_API_VERSION", api))
    else:
        args.append(self.define("DEFAULT_API_VERSION", api))
```